pkg/telemetry/tracing.go

```diff
@@ -0,0 +1,164 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type TracerType int8
+
+const (
+	TracerTypeIO = iota
+	TracerTypeOTLP
+	TracerTypeJaeger
+)
+
+type tracer struct {
+	Provider *trace.TracerProvider
+	Shutdown func()
+}
+
+func TracerSetup(svc string, ttype TracerType) (func(), error) {
+	ctx := context.Background()
+
+	var (
+		tracer *tracer
+		err    error
+	)
+
+	tracer, err = NewTracerProvider(ctx, svc, ttype)
```

Comment:
Is this separate declaration needed?

```suggestion
	tracer, err := NewTracerProvider(ctx, svc, ttype)
```

Reply:
nope. there was a reason but don't remember why.